src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -740,13 +740,16 @@ async def get_exporting_owner(owned_dependency: OwnedDependency) -> ExportedTarg
                 sibling = next(exported_ancestor_iter, None)
             if sibling_owners:
                 raise AmbiguousOwnerError(
-                    f"Exporting owners for {target.address} are "
-                    f"ambiguous. Found {exported_ancestor.address} and "
-                    f"{len(sibling_owners)} others: "
+                    f"python_distribution target owning {target.address} is ambiguous. "
+                    f"Found {exported_ancestor.address} and {len(sibling_owners)} others: "
```

Comment:
What is the remedy? Something like this?

> To fix, refactor {target.address} so that ...

Or even

> See https://www.pantsbuild.org/v2.0/docs/python-setup-py-goal for more information on how the owner is calculated for targets.

Imo we want a user to walk away from every error message with some idea of what to do. This is why Rust and Elm error messages are so excellent.

Bonus would be to explain why this ambiguity is an issue. As I mentioned in Slack, I personally found a concise explanation of why this error exists in the first place insightful to make Pants's algorithm seem less finicky. It helps shift the mindset from Pants being this annoying thing to satisfy, to instead being "your friend" looking out for you, like when people say "the Rust compiler is your friend".

Reply:
Updated and also added link to the docs page.